wlauto/workloads/uxperf_applaunch/__init__.py

```diff
@@ -0,0 +1,227 @@
+
+    def clean_application(self):
+        """Kills the application and clears caches if required
+        """
+        self.stop_application()
+        if(self.applaunch_type == 'launch_from_long-idle'):
+            self.drop_inodepagecache()
+
+    def update_iteration_result(self, context, i):
+        """Creates subfolders for every iterations and respective log files
+        """
+        iteration_folder = "{0}_{1}_{2}".format(
+            self.workload.name, i, self.applaunch_iterations)
+        iteration_path = os.path.join(
+            context.output_directory, iteration_folder)
+        if not os.path.exists(iteration_path):
+            os.makedirs(iteration_path)
+        logcat_log = os.path.join(iteration_path, i + '_logcat.log')
+        self.device.dump_logcat(logcat_log)
+
+    def update_result(self, context):
+        self.logcat_log = open(os.path.join(
+            context.output_directory, 'logcat.log'), 'w')
+        marker = 'UX_PERF'
+        for iter_dir in os.listdir(context.output_directory):
+            if self.workload.name in iter_dir:
+                dir_path = os.path.join(context.output_directory, iter_dir)
+                for log_file in os.listdir(dir_path):
+                    if log_file.endswith('.log'):
+                        log_number = "applaunch{}".format(
+                            log_file.split("_")[0])
+                        file_path = os.path.join(dir_path, log_file)
+                        with open(file_path, 'r') as log_data:
+                            for line in log_data.readlines():
+                                if marker in line and log_number in line:
+                                    self.logcat_log.write(line)
+        self.logcat_log.close()
+        context.add_iteration_artifact(name='logcat',
+                                       path='logcat.log',
+                                       kind='log',
+                                       description='Logcat dump for the run.')
+
+    def init_run(self, context):
+        """Run the setup run for applaunch, clear first time dialogues
+        """
+        self.workload.uiauto_method = "runApplaunchSetup"
+        UiAutomatorWorkload.setup(self.workload, context)
+        UiAutomatorWorkload.run(self.workload, context)
+
+    def metric_run(self, context):
+        """ Run iterations of applaunch test
+        """
+        for iteration in xrange(self.applaunch_iterations):
+            if self.applaunch_type != 'launch_from_background':
+                self.clean_application()
+            sleep(10)
+            self.workload.uiauto_method = "runApplaunchIteration"
+            self.workload.uiauto_params['iteration_count'] = iteration
+            UiAutomatorWorkload.setup(self.workload, context)
+            AndroidBenchmark.clean_process(self.workload, context)
+            UiAutomatorWorkload.run(self.workload, context)
+            self.update_iteration_result(context, str(iteration))
```

Comment:
The initialization and processing should be performed outside of the main run.
If some setup is required between runs, e.g. killing background tasks, it should be done within the java code as this will result in less overhead.